### PR TITLE
fix(material/slider): form control w/ initial disabled state

### PR DESCRIPTION
**Pull request by wagnermaciel**

### CODE REVIEW

File: src/material/slider/slider-input.ts [hunk 1]
```diff
@@ -275,7 +275,12 @@ export class MatSliderThumb implements _MatSliderThumb, OnDestroy, ControlValueA
   /** @docs-private */
   initProps(): void {
     this._updateWidthInactive();
-    this.disabled = this._slider.disabled;
+
+    // If this or the parent slider is disabled, just make everything disabled.
+    if (this.disabled !== this._slider.disabled) {
+      this._slider.disabled = true;
```
Comment (devversion) [inline src/material/slider/slider-input.ts:281]: should this just happen on init? and not when the setter for `disabled` triggers?

**Review by devversion — APPROVED** [review]

File: src/material/slider/slider-input.ts (hunk 1 above)
Comment (wagnermaciel) [reply src/material/slider/slider-input.ts:281]: This behavior is already present in the `disabled` setters. If either slider thumb gets disabled or enabled, the entire slider is updated to that state. On init the handling was incorrect bc it didn't consider the fact that the slider inputs could be set to disabled before this point in time

Comment (devversion) [reply src/material/slider/slider-input.ts:281]: Makes sense, thx for the explanation.

**Review by devversion — APPROVED** [review]

File: src/material/slider/slider-input.ts (hunk 1 above)
Comment (wagnermaciel) [reply src/material/slider/slider-input.ts:281]: No problem. For more context, this specific issue happens when someone initializes a form control w/ disabled true

```
control = new FormControl({value: '0', disabled: true});
```

**Pull request merged**